application/backend/app/services/pipeline_service.py

```diff
@@ -31,8 +34,21 @@ def create_pipeline(self, project_id: UUID) -> Pipeline:
     def get_active_pipeline(self) -> Pipeline | None:
         """Retrieve an active pipeline."""
         pipeline_repo = PipelineRepository(self._db_session)
-        pipeline = pipeline_repo.get_active_pipeline()
-        return Pipeline.model_validate(pipeline) if pipeline is not None else None
+        pipeline_db = pipeline_repo.get_active_pipeline()
+        if pipeline_db is None:
+            return None
+
+        pipeline = Pipeline.model_validate(pipeline_db)
```

Comment:
This line can be placed below the `if` block, so the extra update on L48 can be omitted.